File: easy/BestTimetoBuyandSellStock.cpp
#include <climits>
#include <vector>

using namespace std;

class Solution {
public:
  int maxProfit(vector<int> &prices) {

    int l = 0;
    int mx_profit = 0;
    for (int i = 1; i < prices.size(); i++) {
      if (prices[l] <= prices[i]) {
        mx_profit = max(mx_profit, prices[i] - prices[l]);

      } else {
        l = i;
      }
    }
    return mx_profit;
  }
};

int main() {}

File: stonycodes70proplems/easy/BestTimetoBuyandSellStock.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # Input: prices = [7,1,5,3,6,4]
        # Output: 5
        l, r = 0, 1
        ans = 0
        while r < len(prices):
            if prices[l] > prices[r]:
                l = r
            else:
                ans = max(ans, prices[r] - prices[l])
                r += 1

        return ans
